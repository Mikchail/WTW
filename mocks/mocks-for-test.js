

const movies = [{
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: true,
}, {
  id: 3,
  title: `Macbeth`,
  cardImg: `../../img/macbeth.jpg`,
  genre: `Drama`,
  year: 2002,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 2,
  ratingCount: 240,
  description: `In the 1930s`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`],
  preview: `https://480p.mp4`,
  videoLink: `https://480p.mp4`,
  runTime: 200,
  isFavorite: true,
}, {
  id: 4,
  title: `Aviator`,
  cardImg: `../../img/aviator.jpg`,
  genre: `Comedy`,
  year: 1988,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 7,
  ratingCount: 188,
  description: `in her murder.`,
  director: `Leo Dicaprio`,
  starring: [`Bill Murray`],
  preview: `400p.ogv.360p.webm`,
  videoLink: `mp4`,
  runTime: 188,
  isFavorite: true,
}];
const comments = [{
  "id": 1,
  "user": {
    "id": 4,
    "name": `Kate Muir`
  },
  "rating": 8.9,
  "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  "date": `2019-05-08T14:13:56.569Z`
}, {
  "id": 2,
  "user": {
    "id": 4,
    "name": `Kate Muir`
  },
  "rating": 8.9,
  "comment": `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  "date": `2019-05-08T14:13:56.569Z`
}];
const newMovie = {
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: false,
};


const newMovies = [{
  id: 2,
  title: `Bohemian Rhapsody`,
  cardImg: `../../img/bohemian-rhapsody.jpg`,
  genre: `Drama`,
  year: 2000,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 5,
  ratingCount: 240,
  description: `In the 1930s, theher murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`],
  preview: `https://upload.wikimedia.org`,
  videoLink: `https://download.blender4`,
  runTime: 99,
  isFavorite: false,
}, {
  id: 3,
  title: `Macbeth`,
  cardImg: `../../img/macbeth.jpg`,
  genre: `Drama`,
  year: 2002,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 2,
  ratingCount: 240,
  description: `In the 1930s`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`],
  preview: `https://480p.mp4`,
  videoLink: `https://480p.mp4`,
  runTime: 200,
  isFavorite: true,
}, {
  id: 4,
  title: `Aviator`,
  cardImg: `../../img/aviator.jpg`,
  genre: `Comedy`,
  year: 1988,
  backgroundImg: `img/bg-the-grand-budapest-hotel.jpg`,
  posterImg: `img/the-grand-budapest-hotel-poster.jpg`,
  rating: 7,
  ratingCount: 188,
  description: `in her murder.`,
  director: `Leo Dicaprio`,
  starring: [`Bill Murray`],
  preview: `400p.ogv.360p.webm`,
  videoLink: `mp4`,
  runTime: 188,
  isFavorite: true,
}];
const recivedMovies = [
  {
    "actors": undefined,
    "backgroundColor": undefined,
    "description": undefined,
    "genre": undefined,
    "id": undefined,
    "isFavorite": undefined,
    "movieCoverSrc": undefined,
    "numberVotes": undefined,
    "posterSrc": undefined,
    "previewVideoLink": undefined,
    "producer": undefined,
    "rating": undefined,
    "runTime": undefined,
    "screenshotSrc": undefined,
    "title": undefined,
    "videoSrc": undefined,
    "yearRelease": undefined,
  },
];

export {movies , newMovie,newMovies,recivedMovies,comments};
