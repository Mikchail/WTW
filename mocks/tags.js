const tags = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export {tags};
