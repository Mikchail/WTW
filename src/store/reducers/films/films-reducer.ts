import { extend } from "../../../utils";
import { GET_FILMS, LOADED_FILMS } from "../../actions/films-actions";

const initialState = {
  films: [],
  isLoading: false,
  isError: false,
}

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_FILMS:
      return state;
    case LOADED_FILMS:
      return extend(state, {films: action.payload, isLoading: true});
    default:
      return state;
  }
};
