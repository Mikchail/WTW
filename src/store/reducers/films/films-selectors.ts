import NameSpace from '../../name-space';
import {createSelector} from 'reselect';
import { RootState } from '../root-reducer';

export const getIsLoading = (state: RootState) => {
  return state[NameSpace.FILMS].isLoading;
};

export const getFilmById = (state: RootState, props: {selectedID: number}) => {
  const film = getFilms(state).find((it) => it.id === props.selectedID);
  return film;
};

export const getSimilarFilms = (state: RootState, id: number) => {
  const filmById = getFilmById(state, {selectedID: id});
  const films = getFilms(state).filter((film) => {
    return Boolean(film.genre === filmById.genre && film.id !== id);
  });
  return films;
};

export const getFilms = (state: RootState) => {
  return state[NameSpace.FILMS].films;
};

export const getTags = createSelector(getFilms, (films) => {
  return [
    ...new Set(
      films.map((f) => {
        return f.genre;
      })
    ),
  ];
});
export const getSelectFilm = (state: RootState, id: number) => {
  return state[NameSpace.FILMS].films.find((it) => it.id === id);
};
