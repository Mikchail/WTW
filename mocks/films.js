import { getRandomNumber, getRandomElement, encodeText } from '../src/utils';
import { tags as genres } from './tags';
import { articles } from './articles';

const titles = articles.map((item) => {
    return item.title;
});
const stringForDescription = [
    `Beautiful & luxurious apartment at great location`,
    `Nice, cozy, wood and stone place`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
    `Beautiful & luxurious apartment at great location`,
    `Nice, cozy, wood and stone place`,
    `Wood and stone place`,
    `Nice, cozy, warm big bed apartment`,
];

const randomDate = () => {
    const min = new Date(1997, 2, 2).getTime();
    const max = new Date(2018, 2, 2).getTime();
    const r = getRandomNumber(min, max);
    return new Date(r);
};

const getMovies = (titles) => {
    return titles.reduce((acc, movie) => {
        console.log(acc);
        acc.push({
            title: movie,
            genre: getRandomElement(genres),
            rating: Math.floor(Math.random() * 5),
            descriptin: `${getRandomElement(
        stringForDescription
      )} ${getRandomElement(stringForDescription)} ${getRandomElement(stringForDescription)}`,
            year: randomDate(),
            src: encodeText(movie),
        });
        return acc;
    }, []);
};
const films = getMovies(titles);
export { films };