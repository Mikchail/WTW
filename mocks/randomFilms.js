import {getRandomNumber, getRandomElement, encodeText} from '../src/utils';
import {tags as genres} from './tags';
import {articles} from './articles';

const titles = articles.map((item) => {
  return item.title;
});

const stringForDescription = [
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`,
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`,
];

const randomDate = () => {
  const min = new Date(1997, 2, 2).getTime();
  const max = new Date(2018, 2, 2).getTime();
  const r = getRandomNumber(min, max);
  return new Date(r);
};

const getMovies = (array) => {
  return array.reduce((acc, movie, index) => {
    acc.push({
      id: index,
      title: movie,
      srcMovie: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      genre: getRandomElement(genres),
      rating: Math.floor(Math.random() * 5),
      countRating: 240,
      descriptin: `${getRandomElement(stringForDescription)} ${getRandomElement(
        stringForDescription
      )} ${getRandomElement(stringForDescription)}`,
      year: randomDate().getFullYear(),
      src: `/img/${encodeText(movie)}.jpg`,
      director: `Wes Andreson`,
      starring: [
        `Bill Murray`,
        `Edward Norton`,
        `Jude Law`,
        `Willem Dafoe`,
        `and other`,
      ],
      reviews: [
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one`,
          date: `December 24, 2016`,
          rating: `8.9`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir one one`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two`,
          date: `December 24, 2016`,
        },
        {
          description: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of
                        one of the director's funniest and most exquisitely designed movies in years.`,
          author: `Kate Muir two two`,
          date: `December 24, 2016`,
        },
      ],
    });
    return acc;
  }, []);
};
const films = getMovies(titles);
export {films};
