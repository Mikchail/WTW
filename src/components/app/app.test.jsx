import React from 'react';
import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { films } from '../../../mocks/films';
import App from './app';
import NameSpace from '../../store/name-space';

const mockStore = configureStore([]);
describe(`App`, () => {
  it(`Render pageMain`, async () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        films,
        isLoading: false,
      },
      [NameSpace.SHOW]: {
        currentGenre: `All genres`,
        filmsByGenre: [],
        sameFilms: [],
        selectedFilm: false,
        favoriteFilms: [],
      },
      [NameSpace.USER]: {
        user: false,
        isLogin: false,
      },
      [NameSpace.FILMS]: {
        films: [],
        isLoading: false,
      },
      [NameSpace.SHOWN_FILM]: {
        currentGenre: ""
      },
      [NameSpace.PROMO_FILM]: {
        filmPromo: null
      }
    });

    render(
      <Provider store={store}>
        <App onFilmSelect={() => { }} />
      </Provider>,
    )

    const mainPage = screen.getByTestId("main-page")
    // const signPage = screen.getByTestId("signin-page")
    const linkSign = screen.getByTestId("link-signin")
    const elem = screen.getByText(/WTW/i)
    expect(elem).toBeInTheDocument()
    userEvent.click(linkSign)
    const signPage = await screen.findByTestId("signin-page")
    expect(signPage).toBeInTheDocument()
  });
});
