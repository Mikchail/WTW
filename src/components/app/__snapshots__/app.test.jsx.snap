// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render pageMain 1`] = `
<div
  style={
    Object {
      "background": "black",
      "height": "100vh",
    }
  }
>
  <div
    className="loader"
  >
    <div
      className="l_main"
    >
      <div
        className="l_square"
      >
        <span />
        <span />
        <span />
      </div>
      <div
        className="l_square"
      >
        <span />
        <span />
        <span />
      </div>
      <div
        className="l_square"
      >
        <span />
        <span />
        <span />
      </div>
      <div
        className="l_square"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
</div>
`;
